scripts/zkai_attention_kv_proof_route_selector_gate.py

```
@@ -535,6 +595,30 @@ def load_stwo_native_masked_sequence_envelope(
+    """Load and validate the current implementation-exact quantized Softmax receipt payload."""
+
+    raw = read_bounded_text(
+        path,
+        QUANTIZED_SOFTMAX_RECEIPT_MAX_JSON_BYTES,
+        "quantized Softmax receipt evidence",
+    )
+    try:
+        payload = json.loads(raw)
+        QUANTIZED_SOFTMAX.validate_result(payload)
+    except QUANTIZED_SOFTMAX.QuantizedSoftmaxReceiptGateError as error:
+        raise AttentionKvRouteSelectorError(f"quantized Softmax receipt drift: {error}") from error
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)